back/engines/commercial/insights/app/services/insights/category_detection_service.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Insights
+  class CategoryDetectionService
+    attr_reader :nlp_client
+
+    def initialize(nlp_client = nil)
+      @nlp_client = nlp_client || NLP::Api.new
+    end
+
+    # @return[Array<Insights::ZeroshotClassificationTask>]
+    def detect_categories(view)
+      inputs = view.scope.ideas
+
+      return [] unless inputs.any?
+
+      detected_categories_attributes = @nlp_client
+        .project_tag_suggestions(
+          top_locale(inputs),
+          AppConfiguration.instance.id,
+          view.scope.id,
+          25
+        )
```

Comment:
Using only a single locale is a huge shortcut IMO. If we stick with it, we should at least inform product about it.

Reply:
Handling multiple languages with simple juxtaposition seemed a lot of effort for a sup-par result, and I think to really support multilingual projects we will need a better holistic approach to multilingual processing from nlp to the front-end, so keeping this shortcoming for now and communicating with product indeed.